### ISSUE TITLE
clang should not warn -Wshorten-64-to-32 when it knows shorten is safe

### ISSUE THREAD

**Issue by rockeet** [llvm/llvm-project#164959]
```c++
#include <limits.h>
unsigned int cvt(unsigned long x) {
    [[assume(x <= UINT_MAX)]];
    return x;
}
```
shorten is safe with `[[assume(x <= UINT_MAX)]];`.
see https://godbolt.org/z/aoPjxco5r
